core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -206,7 +208,7 @@ public Schema apply() {
 
   @Override
   public void commit() {
-    TableMetadata update = applyChangesToMapping(base.updateSchema(apply(), lastColumnId));
+    TableMetadata update = applyChangesToMapping(base.updateSchema(apply(), lastColumnId, lastPartitionFieldId));
```

Comment:
Why is the last field ID passed here?

Reply:
yeah, its not required.
As its a schema update, and ideally we dont need keep lastPartitionFieldId at the schemaUpdate level.